src/CoreFoundation/CFProxySupport.cs

```diff
@@ -579,62 +581,241 @@ public static CFProxySettings GetSystemProxySettings ()
+				var runLoop = CFRunLoop.Current;
+				CFProxyAutoConfigurationResultCallbackInternal cb = delegate (IntPtr client, IntPtr proxyList, IntPtr error) {
+					proxies = ParseProxies (proxyList);
+					if (error != IntPtr.Zero)
+						outError = new NSError (error);
+					runLoop.Stop ();
+				};
+				var clientContext = new CFStreamClientContext ();
+				using (var loopSource = new CFRunLoopSource (factory (cb, ref clientContext)))
+				using (var mode = new NSString ("Xamarin.iOS.Proxy")) {
+					runLoop.AddSource (loopSource, mode);
+					// blocks until stop is called, will be done in the cb set previously
+					runLoop.RunInMode (mode, double.MaxValue, false);
+					runLoop.RemoveSource (loopSource, mode);
+				}
+			}, cancellationToken).ConfigureAwait (false);
+
+			return (proxies: proxies, error: outError);
+		}
+
+		static CFProxy[] ExecutePacCFRunLoopSourceBlocking (CreatePACCFRunLoopSource factory, out NSError outError)
+		{
+			outError = null;
+			IntPtr cbError = IntPtr.Zero;
+			CFProxy[] proxies = null;
+
+			var runLoop = CFRunLoop.Current;
+			CFProxyAutoConfigurationResultCallbackInternal cb = delegate (IntPtr client, IntPtr proxyList, IntPtr error) {
+				proxies = ParseProxies (proxyList);
+				cbError = error;
+				runLoop.Stop ();
+			};
+
+			var clientContext = new CFStreamClientContext ();
+			using (var loopSource = new CFRunLoopSource (factory (cb, ref clientContext)))
+			using (var mode = new NSString ("Xamarin.iOS.Proxy")) {
+				runLoop.AddSource (loopSource, mode);
+				runLoop.RunInMode (mode, double.MaxValue, false);
+				runLoop.RemoveSource (loopSource, mode);
+				if (cbError != IntPtr.Zero)
+					outError = new NSError (cbError);
+				return proxies;
+			}
+		}
+
+		static CFRunLoopSource ExecutePacCFRunLoopSourceBlocking (CreatePACCFRunLoopSource factory, CFProxyAutoConfigurationResultCallback clientCb, CFStreamClientContext clientContext)
+		{
+			CFProxyAutoConfigurationResultCallbackInternal cb = delegate (IntPtr client, IntPtr proxyList, IntPtr error) {
+				var proxies = ParseProxies (proxyList);
+				clientCb ((client != IntPtr.Zero)? null : new NSObject (client), proxies, (error == IntPtr.Zero) ? null : new NSError (error));
+			};
+			var loopSourcePtr = factory (cb, ref clientContext);
+			return (loopSourcePtr == IntPtr.Zero) ? null : new CFRunLoopSource (loopSourcePtr);
+		}
+
 		[DllImport (Constants.CFNetworkLibrary)]
 		extern static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationScript (
 			/* CFStringRef __nonnull */ IntPtr proxyAutoConfigurationScript,
 			/* CFURLRef __nonnull */ IntPtr targetURL,
-			/* CFProxyAutoConfigurationResultCallback __nonnull */ IntPtr cb,
-			/* CFStreamClientContext * __nonnull */ IntPtr clientContext);
-		
-		public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString proxyAutoConfigurationScript, NSUrl targetURL, CFProxyAutoConfigurationResultCallback resultCallback, CFStreamClientContext clientContext)
```

Comment:
^ `public` ? breaking change (or non public type?)

Reply:
@spouliot it was ignored in the #if false so it was never exposed.